### PR TITLE
Add report component

### PR DESCRIPTION
**Pull request by mantuok**
# Description

Add a "Report" component to display hazard information according to the current design solution. The component includes hazard cards and address details with placeholder for future share functionality. Adjusted for responsive layouts and integrated mock data, with all unit tests passing.

#82 

## Type of changes
- [ ] Bugfix
- [ ] Chore
- [x] New Feature

## Testing
- [ ] I added automated tests
- [x] I think tests are unnecessary

https://github.com/user-attachments/assets/c9984f65-fb40-40bf-823d-540732f039f8

### CODE REVIEW

File: compose.yaml
Comment (agennadi) [inline compose.yaml:1]: I made a typo on line 41, could you please remove the space in "db-data:/var/lib/postgr esql/data"?

File: .gitignore [hunk 1]
```diff
@@ -22,4 +22,5 @@ node_modules
 
 # Sensitive config
 .env.development.local
-.env
\ No newline at end of file
+.env
+.env.local
```
Comment (nickvisut) [inline .gitignore:26]: 👍

File: app/components/card-hazard.tsx [hunk 2]
```diff
@@ -22,8 +31,8 @@ const CardHazard = () => {
         }}
       >
         <HStack justifyContent="space-between">
-          <Text textStyle="textBig">{hazard.title}</Text>
-          <Pill />
+          <Text textStyle="textBig">{title}</Text>
+          <Pill name={hazard.name} />
```
[the remark this answers is not in the document]
Comment (nickvisut) [reply app/components/card-hazard.tsx:35]: e.g., 
```suggestion
          <Pill name={name} />
```

File: app/components/pill.tsx [hunk 3]
```diff
@@ -15,15 +19,17 @@ const Pill = () => {
     }
   };
 
+  const status = AddressData[name as keyof AddressDataType] || "No data";
+
   return (
     <Box>
       <Text
-        bgColor={getBgColor(AddressData.softStory)}
+        bgColor={getBgColor(status)}
```
Comment (nickvisut) [inline app/components/pill.tsx:27]: I liked the separating the hazard data from the colors for the hazard card above. Is it preferred to do it differently here?

File: app/components/card-hazard.tsx [hunk 4]
```diff
@@ -6,14 +6,23 @@ import {
...
+    name: string;
+    title: string;
+    description: string;
+    update: string;
+    color: string;
+  };
+}
 
+const CardHazard: React.FC<CardHazardProps> = ({ hazard }) => {
+  const { title, description, update, color } = hazard;
```
Comment (nickvisut) [inline app/components/card-hazard.tsx:23]: Nit: If you won't use `hazard` directly (see other comment below), you could consider destructuring in the parameters

File: app/page.tsx [hunk 5]
```diff
@@ -22,8 +23,7 @@ const Home = () => {
         </Box>
       </Box>
       <Box
-        w={{ base: "base", md: "741px", xl: "xl" }}
-        h={{ base: "323px", md: "411px", xl: "462px" }}
+        w={{ base: "base", xl: "xl" }}
```
Comment (nickvisut) [inline app/page.tsx:26]: is it confirmed we won't be using the width and height from the current spec?

File: compose.yaml
Comment (nickvisut) [inline compose.yaml:1]: I believe this was addressed when @mantuok resolved merge conflicts (shouldn't have been showing in the diff, anyway) @agennadi

**Comment by nickvisut** [comment]
Looks great! Added a few comments. Will also test locally momentarily.

**Review by agennadi — APPROVED** [review]
Looks good! The page is a little different from the Figma design. I assume this is intentional?

File: app/page.tsx (hunk 5 above)
Comment (nickvisut) [reply app/page.tsx:26]: Something along these lines, but you probably need to adjust at least the padding between:
```tsx
<Box
  w={{ base: "base", md: "741px", xl: "xl" }}
  p={{
    base: "23px 24px 8px 24px",
    md: "37px 27px 8px 26px",
    xl: "50px 128px 8px 127px",
  }}
  m="auto"
>
  <Report />
</Box>
<Box
  w={{ base: "base", md: "741px", xl: "xl" }}
  h={{ base: "323px", md: "411px", xl: "462px" }}
  p={{
    base: "23px 24px 27px 24px",
    md: "37px 27px 28px 26px",
    xl: "50px 128px 40px 127px",
  }}
  m="auto"
>
  <Box
    h="100%"
    border="1px solid"
    borderColor="grey.400"
    borderRadius="8px"
    boxShadow="0px 2px 4px -1px #0000000F, 0px 4px 6px -1px #0000001A"
    overflow="hidden"
  >
    <Map />
  </Box>
```

**Review by nickvisut — CHANGES_REQUESTED** [review]

**Comment by nickvisut** [comment]
> Looks great! Added a few comments. Will also test locally momentarily.

@mantuok found one blocker in `page.tsx`, but other than that, looks really nice!

**Comment by mantuok** [comment]
> Looks good! The page is a little different from the Figma design. I assume this is intentional?

Yes, this is the original design, Micah has recently changed it, but the changes are not confirmed yet, so we agreed to stick with the original for now :)

File: app/components/card-hazard.tsx (hunk 4 above)
Comment (mantuok) [reply app/components/card-hazard.tsx:23]: Good point—updated to destructure the parameter as suggested.

File: app/page.tsx (hunk 5 above)
Comment (mantuok) [reply app/page.tsx:26]: I've left the map out of the report and kept its width and height parameters as they were originally done.

File: app/components/pill.tsx (hunk 3 above)
Comment (mantuok) [reply app/components/pill.tsx:27]: Thanks for pointing that out! I’ll keep it like this for now, but we can revisit once the updated design is confirmed.

**Review by nickvisut — APPROVED** [review]

**Pull request merged**